io.gitlab.hamadmarri.gamma.json

```diff
@@ -0,0 +1,45 @@
+{
+	"app-id": "io.gitlab.hamadmarri.gamma",
+	"runtime": "org.gnome.Platform",
+	"runtime-version": "3.34",
+	"sdk": "org.gnome.Sdk",
+	"command": "gamma",
+	"finish-args": [
+		"--share=network",
+		"--socket=x11",
+		"--socket=wayland",
+		"--share=ipc",
+		"--socket=session-bus",
```

Comment:
You should run the app with flatpak run --log-session-bus io.gitlab.hamadmarri.gamma and just add whatever dbus access you need.

Reply:
I got this output

```
flatpak-builder --force-clean ./build io.gitlab.hamadmarri.gamma.json && flatpak-builder --run --log-session-bus ./build io.gitlab.hamadmarri.gamma.json gamma
Emptying app dir './build'
Downloading sources
warning: redirecting to https://gitlab.com/hamadmarri/gamma-text-editor.git/
Fetching git repo https://gitlab.com/hamadmarri/gamma-text-editor, ref refs/heads/flatpak-0.0.2
warning: redirecting to https://gitlab.com/hamadmarri/gamma-text-editor.git/
remote: Total 0 (delta 0), reused 0 (delta 0), pack-reused 0
Starting build of io.gitlab.hamadmarri.gamma
Cache hit for gtksourceview, skipping build
Cache hit for gamma, skipping build
Cache hit for cleanup, skipping
Cache miss, checking out last cache hit
Finishing app
Exporting share/applications/io.gitlab.hamadmarri.gamma.desktop
Exporting share/icons/hicolor/128x128/apps/io.gitlab.hamadmarri.gamma.svg
Exporting share/icons/hicolor/64x64/apps/io.gitlab.hamadmarri.gamma.svg
Please review the exported files and the metadata
Committing stage finish to cache
Pruning cache
C1: -> org.freedesktop.DBus call org.freedesktop.DBus.Hello at /org/freedesktop/DBus
C2: -> org.freedesktop.DBus fake wildcarded AddMatch for org.freedesktop.portal
C3: -> org.freedesktop.DBus fake wildcarded AddMatch for org.mpris.MediaPlayer2.io.gitlab.hamadmarri.gamma
C4: -> org.freedesktop.DBus fake wildcarded AddMatch for io.gitlab.hamadmarri.gamma
C5: -> org.freedesktop.DBus fake ListNames
B1: <- org.freedesktop.DBus return from C1
B2: <- org.freedesktop.DBus signal org.freedesktop.DBus.NameAcquired at /org/freedesktop/DBus
B3: <- org.freedesktop.DBus return from C2
*SKIPPED*
B4: <- org.freedesktop.DBus return from C3
*SKIPPED*
B5: <- org.freedesktop.DBus return from C4
*SKIPPED*
B6: <- org.freedesktop.DBus return from C5
C6: -> org.freedesktop.DBus fake GetNameOwner for org.freedesktop.DBus
C7: -> org.freedesktop.DBus fake GetNameOwner for org.freedesktop.portal.Desktop
C8: -> org.freedesktop.DBus fake GetNameOwner for org.freedesktop.portal.Documents
*SKIPPED*
C9: -> org.a11y.Bus call org.a11y.Bus.GetAddress at /org/a11y/bus
*HIDDEN* (ping)
B7: <- org.freedesktop.DBus return from C6
*SKIPPED*
B8: <- org.freedesktop.DBus return from C7
*SKIPPED*
B9: <- org.freedesktop.DBus return from C8
*SKIPPED*
B10: <- (no sender) return from C9
*REWRITTEN*

(gamma.py:6): dbind-WARNING **: 13:23:34.568: AT-SPI: Error retrieving accessibility bus address: org.freedesktop.DBus.Error.ServiceUnknown: org.freedesktop.DBus.Error.ServiceUnknown
C1: -> org.freedesktop.DBus call org.freedesktop.DBus.Hello at /org/freedesktop/DBus
C2: -> org.freedesktop.DBus fake wildcarded AddMatch for org.freedesktop.portal
C3: -> org.freedesktop.DBus fake wildcarded AddMatch for org.mpris.MediaPlayer2.io.gitlab.hamadmarri.gamma
C4: -> org.freedesktop.DBus fake wildcarded AddMatch for io.gitlab.hamadmarri.gamma
C5: -> org.freedesktop.DBus fake ListNames
B1: <- org.freedesktop.DBus return from C1
B2: <- org.freedesktop.DBus signal org.freedesktop.DBus.NameAcquired at /org/freedesktop/DBus
B3: <- org.freedesktop.DBus return from C2
*SKIPPED*
B4: <- org.freedesktop.DBus return from C3
*SKIPPED*
B5: <- org.freedesktop.DBus return from C4
*SKIPPED*
B6: <- org.freedesktop.DBus return from C5
C6: -> org.freedesktop.DBus fake GetNameOwner for org.freedesktop.DBus
C7: -> org.freedesktop.DBus fake GetNameOwner for org.freedesktop.portal.Desktop
C8: -> org.freedesktop.DBus fake GetNameOwner for org.freedesktop.portal.Documents
*SKIPPED*
B7: <- org.freedesktop.DBus return from C6
*SKIPPED*
B8: <- org.freedesktop.DBus return from C7
*SKIPPED*
B9: <- org.freedesktop.DBus return from C8
*SKIPPED*
C9: -> org.freedesktop.DBus call org.freedesktop.DBus.AddMatch at /org/freedesktop/DBus
C10: -> org.freedesktop.DBus call org.freedesktop.DBus.StartServiceByName at /org/freedesktop/DBus
Filtering message due to arg0 org.gtk.vfs.Daemon, policy: 0 (required 2)
*HIDDEN* (ping)
B10: <- org.freedesktop.DBus return from C9
B11: <- (no sender) return from C10
*REWRITTEN*
C11: -> org.freedesktop.DBus call org.freedesktop.DBus.GetNameOwner at /org/freedesktop/DBus
Filtering message due to arg0 org.gtk.vfs.Daemon, policy: 0 (required 1)
B12: <- (no sender) return from C11
*REWRITTEN*
C12: -> org.gtk.vfs.Daemon call org.gtk.vfs.Daemon.ListMonitorImplementations at /org/gtk/vfs/Daemon
*HIDDEN* (ping)
B13: <- (no sender) return from C12
*REWRITTEN*
C1: -> org.freedesktop.DBus call org.freedesktop.DBus.Hello at /org/freedesktop/DBus
C2: -> org.freedesktop.DBus fake wildcarded AddMatch for org.freedesktop.portal
C3: -> org.freedesktop.DBus fake wildcarded AddMatch for org.mpris.MediaPlayer2.io.gitlab.hamadmarri.gamma
C4: -> org.freedesktop.DBus fake wildcarded AddMatch for io.gitlab.hamadmarri.gamma
C5: -> org.freedesktop.DBus fake ListNames
B1: <- org.freedesktop.DBus return from C1
B2: <- org.freedesktop.DBus signal org.freedesktop.DBus.NameAcquired at /org/freedesktop/DBus
B3: <- org.freedesktop.DBus return from C2
*SKIPPED*
B4: <- org.freedesktop.DBus return from C3
*SKIPPED*
B5: <- org.freedesktop.DBus return from C4
*SKIPPED*
B6: <- org.freedesktop.DBus return from C5
C6: -> org.freedesktop.DBus fake GetNameOwner for org.freedesktop.DBus
C7: -> org.freedesktop.DBus fake GetNameOwner for org.freedesktop.portal.Desktop
C8: -> org.freedesktop.DBus fake GetNameOwner for org.freedesktop.portal.Documents
*SKIPPED*
B7: <- org.freedesktop.DBus return from C6
*SKIPPED*
B8: <- org.freedesktop.DBus return from C7
*SKIPPED*
B9: <- org.freedesktop.DBus return from C8
*SKIPPED*
C9: -> org.freedesktop.DBus call org.freedesktop.DBus.AddMatch at /org/freedesktop/DBus
C10: -> org.freedesktop.DBus call org.freedesktop.DBus.StartServiceByName at /org/freedesktop/DBus
Filtering message due to arg0 org.gtk.vfs.Daemon, policy: 0 (required 2)
*HIDDEN* (ping)
B10: <- org.freedesktop.DBus return from C9
B11: <- (no sender) return from C10
*REWRITTEN*
C11: -> org.freedesktop.DBus call org.freedesktop.DBus.GetNameOwner at /org/freedesktop/DBus
Filtering message due to arg0 org.gtk.vfs.Daemon, policy: 0 (required 1)
B12: <- (no sender) return from C11
*REWRITTEN*
C12: -> org.gtk.vfs.Daemon call org.gtk.vfs.MountTracker.ListMountableInfo at /org/gtk/vfs/mounttracker
*HIDDEN* (ping)
B13: <- (no sender) return from C12
*REWRITTEN*
C13: -> org.freedesktop.DBus call org.freedesktop.DBus.RemoveMatch at /org/freedesktop/DBus
B14: <- org.freedesktop.DBus return from C13
C1: -> org.freedesktop.DBus call org.freedesktop.DBus.Hello at /org/freedesktop/DBus
C2: -> org.freedesktop.DBus fake wildcarded AddMatch for org.freedesktop.portal
C3: -> org.freedesktop.DBus fake wildcarded AddMatch for org.mpris.MediaPlayer2.io.gitlab.hamadmarri.gamma
C4: -> org.freedesktop.DBus fake wildcarded AddMatch for io.gitlab.hamadmarri.gamma
C5: -> org.freedesktop.DBus fake ListNames
C1: -> org.freedesktop.DBus call org.freedesktop.DBus.Hello at /org/freedesktop/DBus
C2: -> org.freedesktop.DBus fake wildcarded AddMatch for org.freedesktop.portal
C3: -> org.freedesktop.DBus fake wildcarded AddMatch for org.mpris.MediaPlayer2.io.gitlab.hamadmarri.gamma
C4: -> org.freedesktop.DBus fake wildcarded AddMatch for io.gitlab.hamadmarri.gamma
C5: -> org.freedesktop.DBus fake ListNames
B1: <- org.freedesktop.DBus return from C1
B2: <- org.freedesktop.DBus signal org.freedesktop.DBus.NameAcquired at /org/freedesktop/DBus
B3: <- org.freedesktop.DBus return from C2
*SKIPPED*
B4: <- org.freedesktop.DBus return from C3
*SKIPPED*
B5: <- org.freedesktop.DBus return from C4
*SKIPPED*
B6: <- org.freedesktop.DBus return from C5
C6: -> org.freedesktop.DBus fake GetNameOwner for org.freedesktop.DBus
C7: -> org.freedesktop.DBus fake GetNameOwner for org.freedesktop.portal.Desktop
C8: -> org.freedesktop.DBus fake GetNameOwner for org.freedesktop.portal.Documents
*SKIPPED*
B1: <- org.freedesktop.DBus return from C1
B2: <- org.freedesktop.DBus signal org.freedesktop.DBus.NameAcquired at /org/freedesktop/DBus
B3: <- org.freedesktop.DBus return from C2
*SKIPPED*
B4: <- org.freedesktop.DBus return from C3
*SKIPPED*
B5: <- org.freedesktop.DBus return from C4
*SKIPPED*
B6: <- org.freedesktop.DBus return from C5
C6: -> org.freedesktop.DBus fake GetNameOwner for org.freedesktop.DBus
C7: -> org.freedesktop.DBus fake GetNameOwner for org.freedesktop.portal.Desktop
C8: -> org.freedesktop.DBus fake GetNameOwner for org.freedesktop.portal.Documents
*SKIPPED*
B7: <- org.freedesktop.DBus return from C6
*SKIPPED*
B8: <- org.freedesktop.DBus return from C7
*SKIPPED*
B9: <- org.freedesktop.DBus return from C8
*SKIPPED*
B7: <- org.freedesktop.DBus return from C6
*SKIPPED*
B8: <- org.freedesktop.DBus return from C7
*SKIPPED*
B9: <- org.freedesktop.DBus return from C8
*SKIPPED*
C9: -> org.freedesktop.DBus call org.freedesktop.DBus.RequestName at /org/freedesktop/DBus
B10: <- org.freedesktop.DBus signal org.freedesktop.DBus.NameOwnerChanged at /org/freedesktop/DBus
B10: <- org.freedesktop.DBus signal org.freedesktop.DBus.NameOwnerChanged at /org/freedesktop/DBus
B11: <- org.freedesktop.DBus signal org.freedesktop.DBus.NameAcquired at /org/freedesktop/DBus
B12: <- org.freedesktop.DBus return from C9
B10: <- org.freedesktop.DBus signal org.freedesktop.DBus.NameOwnerChanged at /org/freedesktop/DBus
C10: -> org.freedesktop.DBus call org.freedesktop.DBus.GetId at /org/freedesktop/DBus
B13: <- org.freedesktop.DBus return from C10
C11: -> org.freedesktop.DBus call org.freedesktop.DBus.AddMatch at /org/freedesktop/DBus
C12: -> org.freedesktop.DBus call org.freedesktop.DBus.AddMatch at /org/freedesktop/DBus
C13: -> org.freedesktop.DBus call org.freedesktop.DBus.AddMatch at /org/freedesktop/DBus
C14: -> org.freedesktop.DBus call org.freedesktop.DBus.StartServiceByName at /org/freedesktop/DBus
B14: <- org.freedesktop.DBus return from C11
B15: <- org.freedesktop.DBus return from C12
B16: <- org.freedesktop.DBus return from C13
B17: <- org.freedesktop.DBus return from C14
C15: -> org.freedesktop.DBus call org.freedesktop.DBus.GetNameOwner at /org/freedesktop/DBus
B18: <- org.freedesktop.DBus return from C15
C16: -> :1.112 call org.freedesktop.DBus.Properties.GetAll at /org/freedesktop/portal/desktop
B122: <- :1.112 return from C16
no GAMMA_OPEN_FILE
C9: -> org.freedesktop.portal.IBus call org.freedesktop.DBus.Peer.Ping at /org/freedesktop/IBus
C17: -> org.freedesktop.DBus call org.freedesktop.DBus.AddMatch at /org/freedesktop/DBus
C18: -> org.freedesktop.DBus call org.freedesktop.DBus.GetNameOwner at /org/freedesktop/DBus
C19: -> org.freedesktop.DBus call org.freedesktop.DBus.GetNameOwner at /org/freedesktop/DBus
B10: <- org.freedesktop.DBus return error org.freedesktop.DBus.Error.ServiceUnknown from C9
B19: <- org.freedesktop.DBus return from C17
B20: <- org.freedesktop.DBus return error org.freedesktop.DBus.Error.NameHasNoOwner from C18
B21: <- org.freedesktop.DBus return error org.freedesktop.DBus.Error.NameHasNoOwner from C19
B806: <- :1.17 call org.gtk.Actions.DescribeAll at /io/gitlab/hamadmarri/gamma
B810: <- :1.17 call org.freedesktop.DBus.Properties.GetAll at /io/gitlab/hamadmarri/gamma
C20: -> :1.17 return from B806
C21: -> :1.17 return from B810
B814: <- :1.17 call org.gtk.Actions.DescribeAll at /io/gitlab/hamadmarri/gamma/window/1
C22: -> :1.17 return from B814
```

Note this:
```
(gamma.py:6): dbind-WARNING **: 13:23:34.568: AT-SPI: Error retrieving accessibility bus address: org.freedesktop.DBus.Error.ServiceUnknown: org.freedesktop.DBus.Error.ServiceUnknown
```

I am not sure what to add though